src/Microsoft.OData.Core/PrimitivePayloadValueConverters/PrimitivePayloadValueConverterProvider.cs

```diff
@@ -0,0 +1,29 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="DefaultPrimitiveValueConverter.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.Core.PrimitivePayloadValueConverters
+{
+    using System;
+    using System.Globalization;
+
+    /// <summary>
+    /// Provides the default of primitive payload value converter
+    /// </summary>
+    public static class PrimitivePayloadValueConverterProvider

```

Comment:
As you're using global static variable to store the converter here, all models would shall the same value converter. This would breaks when a service wants to deal with 2+ models with different configurations at the same time.

We can consider leverage model.SetAnnotationValue & model.GetAnnotationValue to make it a per model settings. Please refer to src\Microsoft.OData.Edm\ExtensionMethods::SetPrimitiveValueConverter for more information.

Also the tests run in PrimitiveValuesRoundtripJsonLightTests class would randomly fails as one case would call global SetPrimitivePayloadValueConverter which changes global serialization/deserialization behaviour.